### PR TITLE
IntelFsp2WrapperPkg: Prevent null pointer dereference

### PR DESCRIPTION
**Pull request by makubacki**
# Description

Return from `FspsWrapperInitDispatchMode()` if a buffer allocation fails instead of attempting to dereference the pointer.

- [ ] Breaking change?
- [ ] Impacts security?
- [ ] Includes tests?

## How This Was Tested

- IntelFsp2WrapperPkg build and CI
- Intel physical platform boot with change (FSP dispatch mode)

## Integration Instructions

- N/A

### CODE REVIEW

**Review by apop5 — APPROVED** [review]

**Review by AshrafAliS — APPROVED** [review]

**Comment by makubacki** [comment]
Thanks for the reviews. Can a maintainer with write access review as well?

**Comment by makubacki** [comment]
@mdkinney, this has been open for ~1 month now with some approvals. Can you please help get it through?

**Comment by makubacki** [comment]
While waiting for Mike, other reviewers, your review would be appreciated.

- @cbduggap
- @ChaselChiu
- @chengangc
- @lzeng14
- @nate-desimone
- @tedkuo1

**Review by lzeng14 — APPROVED** [review]

File: IntelFsp2WrapperPkg/FspsWrapperPeim/FspsWrapperPeim.c
Comment (AshrafAliS) [inline IntelFsp2WrapperPkg/FspsWrapperPeim/FspsWrapperPeim.c:436]: Do we need the freepool when memory is not allocated

Comment (lzeng14) [reply IntelFsp2WrapperPkg/FspsWrapperPeim/FspsWrapperPeim.c:436]: It is to free allocated MeasurementExcludedFvPpi, not unallocated MeasurementExcludedPpiList, so should be good.

Comment (AshrafAliS) [reply IntelFsp2WrapperPkg/FspsWrapperPeim/FspsWrapperPeim.c:436]: Thank you

**Review by AshrafAliS — APPROVED** [review]

**Comment by makubacki** [comment]
@mdkinney, several reviewers have approved. Can you please help provide a review with write permission?

**Review by nate-desimone — APPROVED** [review]

**Pull request merged**
